src/server/visual-diff-plugin.js

```diff
@@ -61,9 +102,27 @@ export function visualDiff({ updateGoldens = false } = {}) {
 			const browser = session.browser.name.toLowerCase();
 			const { dir, newName } = extractTestPartsFromName(payload.name);
 			const testPath = dirname(session.testFile).replace(rootDir, '');
-			const goldenFileName = `${join(rootDir, '.vdiff', testPath, 'golden', browser, dir, newName)}.png`;
-			const passFileName = `${join(rootDir, '.vdiff', testPath, 'pass', browser, dir, newName)}.png`;
-			const screenshotFileName = `${join(rootDir, '.vdiff', testPath, 'fail', browser, dir, newName)}.png`;
+			const newPath = join(rootDir, '.vdiff', testPath, dir);
+			const goldenFileName = `${join(newPath, 'golden', browser, newName)}.png`;
+			const passFileName = `${join(newPath, 'pass', browser, newName)}.png`;
+			const screenshotFileName = `${join(newPath, 'fail', browser, newName)}.png`;
+
+			if (!isCI) { // CI will be a fresh .vdiff folder each time and only one run
+				if (session.testRun !== currentRun) {
+					currentRun = session.testRun;
+					clearedDirs.clear();
+				}
+
+				if (runSubset || currentRun > 0) {
+					if (!clearedDirs.has(newPath)) {
+						const promise = clearDir(updateGoldens, newPath);
+						clearedDirs.set(newPath, promise);
+						await promise;
+					} else {
+						await clearedDirs.get(newPath);
+					}
```

Comment:
Nit, slightly less code:
```suggestion
					if (!clearedDirs.has(newPath)) {
						clearedDirs.set(newPath, clearDir(updateGoldens, newPath));
					}
					await clearedDirs.get(newPath);
```